src/modbus.c

```diff
@@ -723,180 +770,84 @@ int modbus_reply(modbus_t *ctx, const uint8_t *req,
     function = req[offset];
     address = (req[offset + 1] << 8) + req[offset + 2];
 
+    /* special RTU-cases error checking */
+    if (ctx->backend->backend_type == _MODBUS_BACKEND_TYPE_RTU) {
+        /* we accept BROADCAST_ADDRESSes */
+        if (slave != MODBUS_BROADCAST_ADDRESS) {
+            /* check whether this slave is handled by this instance and
+             * suppress any responses when the slave-id not accepted by the user */
+            if (ctx->reply_cb->accept_rtu_slave(ctx->reply_user_ctx, slave) == FALSE) {
+                if (ctx->debug)
+                    fprintf(stderr, "slave ID %d is not handled by this instance\n", slave);
+                return 0;
+            }
+        }
+
+		// TODO broadcast responses should use the slave-id, probably
```

Comment:
Broadcasts simply do not have a response by definition.

Reply:
Thanks for helping me with this question.

Reply:
however, it is a quirk that many devices use, and has been porposed as one of the "quirks_modes" that libmodbus should support using....  (sorry for the necro)